### DISCUSSION TITLE
Its safe to call .block() on database query/insert result?

### DISCUSSION THREAD

**Discussion by thankusWR** [pgjdbc/r2dbc-postgresql#432] [Q&A]
In my service I have one use cases when synchronous call to database is required and I wonder if I can securely call `block()` on that result?
Method is called in non-reactive thread so on my tests it works fine but from what heard in some cases its dangerous to call `block()`.

Below is my repository and DB operation on which I want to block.
```
@Repository
public interface UserRepository extends ReactiveCrudRepository<User, Long> {

    Mono<User> findByUserGuid(UUID userGuid);
}
```

```
userRepository
                .findByUserGuid(userGuid)
                .switchIfEmpty(userRepository.save(new User(userGuid)))
                .block()
```

**Comment by mp911de** [comment]
Calling `.block()` does exactly that to the thread that the code is being called on: It blocks. 

> In my service I have one use cases when synchronous call to database is required

This is a perfectly valid application, however, you should pass in a `Duration` to ensure proper timeout handling. And, this is more important, please make sure that the Thread, that calls `.block()` isn't one from a non-blocking thread pool/event loop, otherwise you will prevent your entire application from progressing.
